lib/vdsm/virt/periodic.py

```diff
@@ -376,13 +376,35 @@ def _execute(self):
 
 class VolumeWatermarkMonitor(_RunnableOnVm):
 
+    @classmethod
+    def dispatch(cls, vm):
+        """
+        Monitor vm soon as possible in the executor thread pool.
+        """
+        task = cls(vm, urgent=True)
+
+        if not task.runnable:
+            vm.log.warning("Could not run %s, domain is not ready", task)
+            return
+
+        try:
+            _executor.dispatch(
+                task,
+                timeout=config.getfloat("thinp", "extend_timeout"))
```

Comment:
What’s the purpose of specifying the timeout?

Reply:
The executor will discard the thread after the timeout, and the periodic monitor will retry after the same interval.

In the current periodic monitor we use 1.0 timeout for 2.0 seconds interval, which seems too low for this use case. I plan to unify the timeout to 2.0 in both cases.

Reply:
We cannot prevent the thread from blocking, for example it can block on getVolumeSize() if the storage domain cache is locked in some cases. We don't want to delay the periodic thread for this or other vms because one threads is block on something.

We already use a timeout for the volume watermark monitor, so this keeps the exiting behavior. Do suggest to change it?

Reply:
No, it makes sense.  And I suppose the primary reason of having the value configurable here is to be consistent with the timeout in thinp.py, right?

Reply:
Yes, and the timeout used by VolumeWatermarkMonitor, which should also use this timeout istead of the "irs:vm_watermark_interval" (not done yet).